### DISCUSSION TITLE
How can I refresh the provider_token?

### DISCUSSION THREAD

**Discussion by ishg** [supabase/supabase#2260] [Questions]
I'm building a tool with Supabase (and loving it! 💟 ) and Nuxt and want to integrate with Google APIs. I use Google Auth with Supabase

```javascript
const { session, error } = await this.$supabase.auth.signIn(
  {
    provider: 'google'
  },
  {
    scopes:
      'https://www.googleapis.com/auth/drive https://www.googleapis.com/auth/bigquery.readonly'
  }
)
// Save session to store, for posterity
commit('SET_SESSION', session)
```
Then I get the `provider_token` from the supabase session to pass to my google API calls.
```javascript
// I believe this gets it from local storage
const token = this.$supabase.auth.session().provider_token
```
This works for a while, but once supabase refreshes the session due to token expiration, the new session does not have a provider token. I confirmed this by inspecting local storage and noticing the `provider_token` value missing. 

**Is there a way to tell supabase to also refresh the `provider_token`?**

**Comment by fredguest** [comment]
I just ran into this too and created an issue for it here https://github.com/supabase/supabase-js/issues/248

**Comment by huuduyentinh14020201** [comment]
https://github.com/ten-tai-kho-n-c-a-b-n-tren-GitHub-U/demo-repository
